### ISSUE TITLE
NLLLoss is not supported in SMPC

### ISSUE THREAD

**Issue by LaRiffle** [OpenMined/PySyft#3233]
We'd like support for the negative log likelihood loss in SMPC.

The Negative Log Likelihood Loss (or Cross Entropy Loss)  is the most common setting for classification problems with C classes. It increases as the predicted probability diverges from the actual label.

Details here: https://pytorch.org/docs/stable/nn.html#nllloss

**Comment by codeboy5** [comment]
I would like to take up this issue

**Comment by LaRiffle** [comment]
Done :)

**Comment by codeboy5** [comment]
Hi, @LaRiffle are we looking for Cross Entry Loss (https://pytorch.org/docs/stable/nn.html#crossentropyloss) or NLL Loss (https://pytorch.org/docs/stable/nn.html#nllloss)? 
A slight difference between the two as Cross-Entropy combines softmax and NLL. I can implement both of these if you would like that?

**Comment by LaRiffle** [comment]
Which one would you think is the most classic in classification? :)

**Comment by codeboy5** [comment]
Cross-Entropy :)

**Comment by LaRiffle** [comment]
Let's run for this one then 🚀

**Comment by Syzygianinfern0** [comment]
I would like to work on this. 
@codeboy5 you can ping me over Slack for Colab as @Syzygianinfern0

**Comment by karlhigley** [comment]
The easiest way to collaborate is probably to push WIP code and work from a shared branch.
